### ISSUE TITLE
Question about LineProfileCharacteristicNominal element and Vector child element

### ISSUE THREAD

**Issue by broush-discus** [QualityInformationFramework/qif-community#32]
In the QIF Version 3.0 schema it appears that element LineProfileCharacteristicNominal is required to have child element Vector.  Is/was this intentional?  If yes, could you provide or point me to information to help me determine - given the typical Profile of a Line feature control frame on a drawing or model, what should child element Vector contain?

This xml
            <LineProfileCharacteristicNominal id="34">
                <CharacteristicDefinitionId>33</CharacteristicDefinitionId>
            </LineProfileCharacteristicNominal>
is yielding this validation error
Error: The element 'LineProfileCharacteristicNominal' in namespace 'http://qifstandards.org/xsd/qif3' has incomplete content. List of possible elements expected: 'FeatureNominalIds, SubstituteFeatureAlgorithm, FeatureZoneIds, EntityInternalIds, EntityExternalIds, Name, CharacteristicDesignator, TargetValue, Vector' in namespace 'http://qifstandards.org/xsd/qif3'.

I've reviewed the 3.0 schema and, unless I've missed something, I see the from list of expected possible elements they are all optional except for Vector.

**Comment by TomRKramer** [comment]
The in-line documentation of the Vector element states: "The vector element is a vector perpendicular to the plane
                in which the characteristic curve lies." 
There is typo in that documentation "vector" should be "Vector". I just corrected that in draft QIF 4.0.
The documentation of the Vector element is also shown in the schema browser [](https://qualityinformationframework.github.io/qif3-browser/qif3.html) if you pick the LineProfileCharacteristicNominal element.

**Comment by broush-discus** [comment]
Tom, thanks for your reply.  I read the in-line documentation notes.  But I think the additional info I added to the original post, the text after "This xml", lead you to miss what I think is a key point.  And let me rephrase it.   Given the typical Profile of a Line feature control frame, like included below, displayed as a model PMI annotation or on a 2D drawing (especially in this case), what could/should child element Vector contain?  How can the user know what to specify?  And/or how can the software product that is facilitating the user to document an inspection characteristic requirement determine this?  In my many years of being involved with FAIR and manufacturing inspection documentation, this is the first time I have encountered something that is saying - Profile of a Line characteristic requirement must have vector data to be valid.  If I've missed something over the years, then I apologize for raising this question that shows it.  My original speculation was the vector data was something some down stream software, machine, tool, CMM, whatever needs to do its job and it got tagged as required in QIF for that reason.  Or, maybe just a mistake.  Thanks for your time.  

![image](https://github.com/user-attachments/assets/ee189726-8afc-4141-990d-736cc5b8399f)

**Comment by TomRKramer** [comment]
On drawings showing profile of a line characteristics, such as the one you posted, it is usually obvious which lines are being controlled (the wavy ones that are formed by planes parallel to the view plane intersecting the part in this case) but if you don't have a drawing, it is not obvious. If you have the same surface and line profile description but no drawing, another candidate for the lines being controlled is straight lines on the curvy surface (which would be perpendicular to the view plane of the drawing). To disambiguate that, it is necessary to identify the orientation of the parallel planes whose intersections with the identified surface of the part  form the lines being controlled. The Vector element does that.

**Comment by broush-discus** [comment]
Tom,  thanks for the additional info/reply.

**Comment by TomRKramer** [comment]
Hi Bob -

I'm glad you found the additional reply helpful. I should have put more
detail in my initial reply.

Tom K
